packages/message-predicates/contract-message-predicate/script_asm.rs

```diff
@@ -1,65 +1,52 @@
 use fuel_asm::{op, GTFArgs, RegId};
-use sha2::{Digest, Sha256};
 
-const PROCESS_MESSAGE_FUNCTION_SIGNATURE: &str = "process_message(u64)";
+const PROCESS_MESSAGE_FUNCTION_SIGNATURE: &str = "process_message";
 const BYTES_PER_INSTR: u16 = 4;
 
 // Gets the bytecode for the message-to-contract script
 pub fn bytecode() -> Vec<u8> {
-    //calculate function selector
-    let mut fn_sel_hasher = Sha256::new();
-    fn_sel_hasher.update(PROCESS_MESSAGE_FUNCTION_SIGNATURE);
-    let fn_sel_hash: [u8; 32] = fn_sel_hasher.finalize().into();
+    let mut fn_selector_bytes =
+        fuels::core::codec::encode_fn_selector(PROCESS_MESSAGE_FUNCTION_SIGNATURE);
+    const FN_SEL_BYTES_LEN: u16 = 23; // new encoding: len of the function signature as u64 (8 bytes) + 15 bytes of function signature ("process_message" has 15 characters)
+    assert_eq!(fn_selector_bytes.len() as u16, FN_SEL_BYTES_LEN);
 
     //register names
     const REG_MEMORY_START_PTR: u8 = 0x10;
-    const REG_ASSET_PTR: u8 = REG_MEMORY_START_PTR;
     const REG_DATA_PTR: u8 = 0x11;
-    const REG_DATA_FN_SEL_PTR: u8 = 0x12;
-    const REG_CONTRACT_ADDR_PTR: u8 = 0x13;
-    const REG_FN_SELECTOR_PTR: u8 = 0x14;
-    const REG_MSG_AMOUNT: u8 = 0x15;
+    const REG_CONTRACT_ADDR_PTR: u8 = 0x12;
+    const REG_FN_SELECTOR_PTR: u8 = 0x13;
+    const REG_DATA_FN_SELECTOR_PTR: u8 = 0x14;
+    const REG_DATA_CALLDATA_PTR: u8 = 0x15;
+    const REG_CALLDATA_PTR: u8 = 0x16;
 
     //referenced data start pointer
-    const REF_DATA_START_PTR: u16 = 11 * BYTES_PER_INSTR;
+    const REF_DATA_START_PTR: u16 = 13 * BYTES_PER_INSTR;
 
-    /* The following assembly code is intended to do the following:
```

Comment:
is this comment still relevant at all? Could you maybe replace it with a new summary instead of removing it entirely?

Reply:
I think removed it by accident, I 'll add it back and update it